torchinductor/scheduler.py

```diff
@@ -596,12 +614,16 @@ def flush(self):
         for backend in self.backends.values():
             backend.flush()
 
-    def codegen_extern_call(self, node: ir.ExternKernel):
-        assert isinstance(node, ir.ExternKernel)
+    def codegen_extern_call(self, scheduler_node: ExternKernelSchedulerNode):
+        assert isinstance(scheduler_node, ExternKernelSchedulerNode)
+        node = scheduler_node.node
         self.flush()
-        node.codegen(V.graph.wrapper_code)
-        self.barrier()
-        self.maybe_free_buffers()
+        if type(node) in TemplateKernels and node.get_device() == "cuda":
+            template_codegen(self, scheduler_node)
+        else:
+            node.codegen(V.graph.wrapper_code)
+            self.barrier()
+            self.maybe_free_buffers()
```

Comment:
I think we still might want the self.barrier()/self.maybe_free_buffers().

Reply:
https://github.com/pytorch/torchdynamo/pull/422/files#diff-eee4a893649579118ce96233528ba599634b8e0d8a04c1c516fb9c45008c2cd6R161-R162
At the end of template_codegen(), there will be `self.barrier()/self.maybe_free_buffers()`. I put them in template_codegen() in case we need to iter_fixed_point again later

Reply:
Makes sense, I missed that